python/mxnet/metric.py

```diff
@@ -793,6 +793,8 @@ def update(self, labels, preds):
 
             if len(label.shape) == 1:
                 label = label.reshape(label.shape[0], 1)
+            if len(pred.shape) == 1:
+                pred = pred.reshape(pred.shape[0], 1)
 
```

Comment:
I think we can also revise the `check_label_shapes` above to be `check_label_shapes(labels, preds, True)` to force the labels and preds have the same shape.

Reply:
Is there a reason to set `shape=0` by default?

Reply:
Not sure. I think shape=1 is more suitable to be the default.

Reply:
@piiswrong can we have your input?

Reply:
I think the check_label_shapes function should be redesigned.
1. it should test if labels is a single NDArray instead of a list, and wrap it with a list `labels=[labels]`
2. 0/1 is not pythonic. Use boolean switches

Reply:
Why don't we check both length and shape? I assume they shoud always match.